app/shared/design-system/components/primitives/numericGestureControl.stories.jsx

```diff
@@ -0,0 +1,39 @@
+import {useRef, useEffect} from 'react';
+import {numericGestureControl} from './numericGestureControl.js';
+
+const useNumericGestureControl = ({ref, value, direction}) => {
+  useEffect(() => {
+    const {disconnectedCallback} = numericGestureControl(ref.current, {
+      initialValue: ref.current.value = value,
+      direction: direction,
+      onValueChange: (event) => {
+        event.preventDefault();
+        event.target.value = event.value;
+        event.target.select();
+      }
+    });
+    return () => disconnectedCallback();
+  }, [direction]);
+}
```

Comment:
IMO this should be the export from the implementation. Why do we want to require developers to use a custom hook every time they want to use this gesture control?

By converting the `numericGestureControl` to a hook, you'll also solve potential issues with mutations and state updates, because ultimately that's what you're trying to express in the end.

Reply:
we can have both, a hook and a function, might be useful